src/Microsoft.Azure.WebJobs.Extensions.Dapr/Services/DaprServiceListener.cs

```diff
@@ -120,6 +140,78 @@ Task GetTopicsAsync(HttpContext context)
             return context.Response.WriteAsync(topicListJson);
         }
 
+        /// <summary>
+        /// Warns if any sidecar settings are misconfigured.
+        /// </summary>
+        internal async Task WarnIfSidecarMisconfigured()
+        {
+            var cancellationToken = new CancellationTokenSource(TimeSpan.FromSeconds(10)).Token;
+
+            string resBody;
+
+            try
+            {
+                var res = await this.daprClient.GetAsync($"{this.daprAddress}/v1.0/metadata", cancellationToken);
+                resBody = await res.Content.ReadAsStringAsync();
+                if (res.StatusCode != System.Net.HttpStatusCode.OK)
+                {
+                    throw new Exception($"Failed to query the Metadata API, received status code {res.StatusCode}, response body: {resBody}");
+                }
+            }
+            catch (Exception ex)
+            {
+                this.logger.LogError($"Failed to query the Metadata API, exception: {ex}");
+                return;
+            }
+
+            JsonDocument jsonDocument;
+
+            try
+            {
+                jsonDocument = JsonDocument.Parse(resBody);
+            }
+            catch (Exception ex)
+            {
+                this.logger.LogError($"Failed to deserialize the Metadata API response, exception: {ex}");
+                return;
+            }
+
+            JsonElement root = jsonDocument.RootElement;
+            if (root.TryGetProperty("appConnectionProperties", out JsonElement appConnectionProperties))
+            {
+                // We set the appAddress, so it's safe to assume it's in the format we expect.
+                // appAddress is in the format "http://127.0.0.1:3001"
+                string[] appAddressParts = this.appAddress.Substring("http://".Length).Split(':');
+                string appChannelAddress = appAddressParts[0];
+                int appPort = int.Parse(appAddressParts[1]);
+
+                try
+                {
+                    int port = appConnectionProperties.GetProperty("port").GetInt32();
+                    if (port != appPort)
+                    {
```

Comment:
What happens if both port and appPort are empty? Are we checking this? I think in case of both empty this if statement would not be true and might not be logged?

Reply:
appPort is never empty, the default appPort is 3001.

Reply:
ok, ideally before this check, DAPR_APP_PORT should be patched as per the DAPR.appPort  value configured by user incase of daprTrigger scenario correct?

Reply:
Yes.
If daprd.appPort is present, we can patch DAPR_APP_PORT with the same.
If daprd.appPort is empty, we can skip updating DAPR_APP_PORT

If user has triggers in their app then daprd.appPort should have a value, which will match with DAPR_APP_PORT
If user does not have trigger configured, daprd.appPort should be empty, which won't set DAPR_APP_PORT as well. 

If user has triggers but daprd.appPort is empty, this will log an error message
```
The Dapr sidecar is not configured to listen on a port, but the app server is running on port 3001. This may cause unexpected behavior, see https://aka.ms/azfunc-dapr-app-config-error.
```

If user has no triggers but daprd.appPort has a value, it will result in sidecar being unhealthy and pod crashing.

Reply:
Is there a trigger or binding check is happening?

Reply:
Yes, this code is only invoked when at least one trigger is registered.